search/result.cpp

```diff
@@ -66,7 +66,7 @@ FeatureID const & Result::GetFeatureID() const
 {
 #if defined(DEBUG)
   auto const type = GetResultType();
-  ASSERT(type == RESULT_FEATURE, (type));
+  ASSERT(type == RESULT_FEATURE || type == RESULT_SUGGEST_FROM_FEATURE, (type));
```

Comment:
Потому что мне нужно получить FeatureID для отображения RESULT_SUGGEST_FROM_FEATURE, а старый код не дает это сделать.

Зачем? Потому что Витя после недавнего разговора с ним по этому поводу, сказал, что при интерактивном поиске на карте мне нужно отображать RESULT_SUGGEST_FROM_FEATURE (и сейчас, без моих изменений, к слову, они уже и так отображаются, Витя просто подтвердил, что это правильно).

Если есть веские причины не отображать в интерактивном поиске RESULT_SUGGEST_FROM_FEATURE, а Витя ввел меня в заблуждение - я откачу это изменение.

Ну или можно просто убрать нафиг этот ассерт.

> On Mar 21, 2016, at 18:26, ygorshenin <notifications@github.com> wrote:
> 
> In search/result.cpp <https://github.com/mapsme/omim/pull/2384#discussion_r56841075>:
> 
> > @@ -66,7 +66,7 @@ FeatureID const & Result::GetFeatureID() const
> >  {
> >  #if defined(DEBUG)
> >    auto const type = GetResultType();
> > -  ASSERT(type == RESULT_FEATURE, (type));
> > +  ASSERT(type == RESULT_FEATURE || type == RESULT_SUGGEST_FROM_FEATURE, (type));
> Зачем это нужно? Мы регулярно до добавляем это исключение то убираем. С точки зрения UI все suggests должны отображаться отлично от обычных результатов, и не должны представлять из себя фичи.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2384/files/0c5f2b528de8c080b03b97e0961cdefcc0f8af2f#r56841075>